### PR TITLE
7.x/feature/dash 257 support list of array for ip whitelist

### PR DESCRIPTION
**Pull request by Michel-Settembrino**

### CODE REVIEW

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.install [hunk 1]
```diff
@@ -5,6 +5,28 @@
...
+ */
+function nexteuropa_dashboard_agent_requirements($phase) {
+  $t = get_t();
+  $requirements = array();
+  $requirements['nexteuropa_dashboard_agent_private_file_system'] = array(
+    'title' => $t('Nexteuropa dashboard agent: Check private file system.'),
+  );
+
+  if (empty(variable_get('file_private_path'))) {
+    $requirements['nexteuropa_dashboard_agent_private_file_system']['descritpion'] = 'Private file system is NOT set';
```
Comment (jonhy81) [inline nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.install]: Typo into `description`

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module [hunk 2]
```diff
@@ -79,26 +79,32 @@ function nexteuropa_dashboard_agent_access_callback() {
...
+            // Current IP is in the list of allowed IPs.
+            // Allow the access.
+            return TRUE;
+          }
         }
+        // If the execution reach this line,
+        // this means current IP is not allowed.
+        watchdog(
+          'nexteuropa_dashboard_agent',
+          "Requester IP *** %requester_ip *** is not allowed. Should be in the range *** %allowed_ips ***.",
```
Comment (jonhy81) [inline nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module]: Update message accordingly the changes.

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module [hunk 3]
```diff
@@ -79,26 +79,32 @@ function nexteuropa_dashboard_agent_access_callback() {
...
-              '%allowed_ips' => $allowed_ips,
-            ),
-            WATCHDOG_WARNING
-          );
-          return FALSE;
+        $allowed_ip_list = explode(',', $allowed_ips);
+        // Compare current IP with the list of allowed IPs.
+        foreach ($allowed_ip_list as $allowed_ip) {
+          $allowed_ip_array = explode('-', $allowed_ip);
+          list($lower, $upper) = $allowed_ip_array;
```
Comment (jonhy81) [inline nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module:86]: If the configuration provide a single ip, your $upper_dec ip will be empty, causing the condition of line 90 to fail.

**Review by jonhy81 — COMMENTED** [review]
Hi @Michel-Settembrino , can you give a quick look to my comments?

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module (hunk 3 above)
Comment (Michel-Settembrino) [reply nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module:86]: Good remark @jonhy81 
This is now solved.

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module (hunk 2 above)
Comment (Michel-Settembrino) [reply nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module]: Message updated.

File: nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.install (hunk 1 above)
Comment (Michel-Settembrino) [reply nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.install]: Done.

**Review by jonhy81 — APPROVED** [review]

**Pull request merged**